docs/dev/run_the_etl.rst

```diff
@@ -11,24 +11,182 @@ just want to use already processed data.
+    couples an asset to the function and upstream assets that are used to produce
+    its contents.*
+
+`SDAs <https://docs.dagster.io/concepts/assets/software-defined-assets>`__
+are the computation building blocks in a Dagster project.
+Assets are linked together to form a direct acyclic graph (DAG) which can
+be execute to persist the data created by the assets. In PUDL, each asset
+is a dataframe written to SQLite or parquet files. Assets in PUDL can be
+raw extracted dataframes, partially cleaned tables or fully normalized
+tables.
+
+SDAs are created by applying the ``@asset`` decorator to a function.
+
+**IO Managers:**
+
+    *IO Managers are user-provided objects that store asset outputs
+    and load them as inputs to downstream assets.*
+
+Each asset has an `IO Manager  <https://docs.dagster.io/concepts/io-management/io-managers>`__
+that tells Dagster how to handle the objects returned by the software
+defined asset's underlying function. The IO Managers in PUDL read and
+write dataframes to and from databases and parquet files.
+
+**Resources:**
+
+   *Resources are objects that are shared across the implementations of
+   multiple software-defined assets and that can be plugged in
+   after defining those assets.*
+
+`Resources <https://docs.dagster.io/concepts/resources>`__ are helpful
+for sharing python objects amongst assets.
+For example, PUDL uses a Datastore resource that multiple assets use
+to pull data from PUDL's raw data archives on Zenodo.
+
+Generally, inputs to assets should either be other assets or
+python objects in Resources.
+
+**Jobs**:
+`Jobs <https://docs.dagster.io/concepts/ops-jobs-graphs/jobs>`__
+are preconfigured collections of assets, resources and IO Managers.
+Jobs are the main unit of execution in Dagster.
+
+**Definitions**:
+`Definitions  <https://docs.dagster.io/concepts/code-locations>`__
+are collections of assets, resources, IO managers and jobs that can
+be loaded into dagit and executed.
+
+There are two main Definitions in the PUDL processing pipeline:
+
+1. ``pudl.ferc_to_sqlite`` :doc:`converts the FERC Form 1 DBF/XBRL files <clone_ferc1>`
+   into a single large `SQLite <https://sqlite.org>`__ database so that the data is
+   easier to extract, and so all of the raw FERC Form 1 data is available in a modern
+   format. You must run this script before you can run ``pudl_etl``.
+2. ``pudl.etl`` coordinates the "Extract, Transform, Load" process that processes
    20+ years worth of data from the FERC Form 1 database, dozens of EIA spreadsheets,
    and the thousands of CSV files that make up the EPA CEMS hourly emissions data into
    a clean, well normalized SQLite database (for the FERC and EIA data), and an `Apache
    Parquet <https://parquet.apache.org/>`__ dataset that is partitioned by state and
    year (for the EPA CEMS).
 
+Both definitions have two preconfigured jobs:
+  - ``etl_fast`` processes one year of data
+  - ``etl_full`` processes all years of data
```

Comment:
This is helpful for understanding the difference between jobs and definitions. Seems like Definitions are like the name of a script, and jobs are like the parameters you can add to the script?? Not the best comparison, but...

Reply:
Yeah! Jobs are kind of like the old pudl_etl or ferc_to_sqlite with preconfigued settings.